client/post-editor/index.js

```diff
@@ -15,18 +15,39 @@ import { makeLayout, render as clientRender } from 'controller';
 export default function() {
 	page( '/post', controller.pressThis, siteSelection, sites, makeLayout, clientRender );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
-	page( '/post/:site?/:post?', siteSelection, controller.post, makeLayout, clientRender );
+	page(
+		'/post/:site?/:post?',
+		siteSelection,
+		controller.post,
+		controller.gutenberg,
```

Comment:
`controller.gutenberg` should run before `controller.post` -- do the redirect before starting to show a classic editor for the post, not after.